src/mlpack/tests/ann_layer_test.cpp

```diff
@@ -1847,4 +1836,122 @@ BOOST_AUTO_TEST_CASE(SubviewIndexTest)
   CheckMatrices(outputEnd, subEnd);
 }
 
+/*
+ * Simple Reparametrization module test.
+ */
+BOOST_AUTO_TEST_CASE(SimpleReparametrizationLayerTest)
+{
+  arma::mat input, output, delta;
+  Reparametrization<> module(5);
+
+  // Test the Forward function.
+  input = join_cols(arma::ones<arma::mat>(5, 1) * -20,
+      arma::zeros<arma::mat>(5, 1));
+  module.Forward(std::move(input), std::move(output));
+  BOOST_REQUIRE_LE(arma::accu(output), 1e-5);
+
+  // Test the Backward function.
+  arma::mat gy = arma::zeros<arma::mat>(5, 1);
+  module.Backward(std::move(input), std::move(gy), std::move(delta));
+  BOOST_REQUIRE(arma::accu(delta) != 0); // klBackward will be added.
+}
+
+/**
+ * Reparametrization module stochastic boolean test.
+ */
+BOOST_AUTO_TEST_CASE(ReparametrizationLayerStochasticTest)
+{
+  arma::mat input, outputA, outputB;
+  Reparametrization<> module(5, false);
+
+  input = join_cols(arma::ones<arma::mat>(5, 1),
+      arma::zeros<arma::mat>(5, 1));
+
+  // Test if two forward passes generate same output.
+  module.Forward(std::move(input), std::move(outputA));
+  module.Forward(std::move(input), std::move(outputB));
+
+  CheckMatrices(std::move(outputA), std::move(outputB));
```

Comment:
Technically no need for the `std::move()` in the `CheckMatrices()` call here but it doesn't make a difference, it still works as-is.